src/main/kotlin/com/chattriggers/ctjs/minecraft/libs/renderer/Font.kt

```diff
@@ -0,0 +1,397 @@
+                            if (shadow) 0.25F else 1F,
+                            if (shadow) 0.25F else 1F,
+                            a
+                        )
+                    }
+                }
+            } else {
+                if (character.code > 255) continue
+                if (obfuscated) character = (character.code + RANDOM_OFFSET).toChar()
+
+                drawChar(character, characterData, x, y)
+
+                val charData = characterData[character.code]
+
+                if (strikethrough) drawLine(
+                    Vector2f(0F, charData!!.height / 2f), Vector2f(
+                        charData.width, charData.height / 2f
+                    )
+                )
+                if (underlined) drawLine(
+                    Vector2f(0F, charData!!.height - 15), Vector2f(
+                        charData.width, charData.height - 15
+                    )
+                )
+
+                x += charData!!.width - 2 * MARGIN
+            }
+        }
+        GlStateManager.popMatrix()
+
+        GlStateManager.color(1F, 1F, 1F, 1F)
+        GlStateManager.bindTexture(0)
+    }
+
+    /**
+     * Gets the width of the given text.
+     *
+     * @param text The text to get the width of.
+     * @return The width of the given text.
+     */
+    fun getWidth(text: String): Float {
+        if (!generatedCharData) return 0F
+
+        var width = 0f
+
+        var characterData = regularData
+        val length = text.length
+
+        for (i in 0 until length) {
+            val character = text[i]
+            val previous = if (i > 0) text[i - 1] else '.'
+
+            if (previous == COLOR_INVOKER) continue
+            if (character == COLOR_INVOKER) {
+                when ("0123456789abcdefklmnor".indexOf(text.lowercase()[i + 1])) {
+                    17 -> characterData = boldData
+                    20 -> characterData = italicsData
+                    21 -> characterData = regularData
+                    else -> {}
+                }
+            } else {
+                if (character.code > 255) continue
+
+                val charData = characterData[character.code]
+                width += (charData!!.width - 2 * MARGIN) / 2
+            }
+        }
+
+        return width + MARGIN / 2
+    }
+
+    /**
+     * Gets the height of the given text.
+     *
+     * @param text The text to get the height of.
+     * @return The height of the given text.
+     */
+    fun getHeight(text: String): Float {
+        if (!generatedCharData) return 0F
+
+        var height = 0f
+
+        var characterData = regularData
+        val length = text.length
+
+        for (i in 0 until length) {
+            val character = text[i]
+            val previous = if (i > 0) text[i - 1] else '.'
+
+            if (previous == COLOR_INVOKER) continue
+            if (character == COLOR_INVOKER) {
+                when ("0123456789abcdefklmnor".indexOf(text.lowercase()[i + 1])) {
```

Comment:
Same hereWhy not just compare against the character directly instead of changing it to an offset?